### ISSUE TITLE
Documenting whole modules

### ISSUE THREAD

**Issue by timholy** [JuliaLang/julia#9628]
As discussed in https://github.com/JuliaLang/julia/pull/9447#issuecomment-68713320, it would be nice to be able to document whole modules. For example,
```
help?> Optim
```
should display the `README.md` for the `Optim` package, if nothing more specific has been defined. Someone who wants to define something briefer might want to be able to say something like
```
doc(Optim, """
The exported functions in Optim are:
    - ...
""")
```
after the module has been defined (or, if easier to implement, inside the module).

**Comment by hayd** [comment]
Is there a way to access the github url (it's in the METADATA)? This may also be useful to show/markdown-link-to in a default doc.

**Comment by one-more-minute** [comment]
As soon as I thought about doing this I realised it's already possible:
```julia
julia> module Foo
       @doc "Bazinga" Foo
       end

julia> @doc Foo
  Bazinga
```
The other option would be to support module syntax directly:
```julia
@doc "Bazinga" ->
module Foo ...
```
Which is easy to implement but not necessarily nicer. Would anyone still like the latter syntax given that the former is possible now?

The readme thing is a nice idea anyway, so I'll have a go at that.

**Comment by one-more-minute** [comment]
I implemented the readme thing in 7d073b896a284274e3d4ead8cf59cac61cab3343 so I'll close this for now. But let me know if you still think the modified syntax is necessary.

**Comment by timholy** [comment]
Looking forward to this, thanks so much!
